components/ui/NewsletterModal.tsx

```diff
@@ -0,0 +1,254 @@
+import type { SectionProps } from "$live/types.ts";
+import {
+  BUTTON_VARIANTS,
+  ButtonVariant,
+} from "$store/components/minicart/Cart.tsx";
+import Icon from "$store/components/ui/Icon.tsx";
+import { Runtime } from "$store/runtime.ts";
+import { useSignal } from "@preact/signals";
+import type { JSX } from "preact";
+import { useEffect, useRef } from "preact/compat";
+import { getCookies } from "std/http/mod.ts";
+const subscribe = Runtime.create(
+  "deco-sites/std/actions/vtex/newsletter/subscribe.ts",
+);
+
+export interface INewsletterInputProps {
+  /**
+   * @title Hide input?
+   */
+  show?: boolean;
+  /**
+   * @title placeholder
+   */
+  placeholder?: string;
+}
+
+export interface INewsletterFormProps {
+  email: INewsletterInputProps;
+  name: INewsletterInputProps;
+  button: {
+    /**
+     * @title button variant
+     * @default primary
+     */
+    variant?: ButtonVariant;
+    /**
+     * @title button label?
+     * @default cadastrar
+     */
+    label?: string;
+  };
+}
+
+export interface Props {
+  /**
+   * @title Newsletter Form
+   */
+  form: INewsletterFormProps;
+  /**
+   * @title newsletter message text?
+   * @format html
+   */
+  text: string;
+
+  /**
+   * @title Days to reopen modal if it is registered
+   */
+  modalSignExpiredDate: number;
+
+  /**
+   * @title Days to reopen moda if it is closed
+   */
+  modalCloseExpiredDate: number;
+}
+
+interface InputNewletterProps {
+  name: string;
+  placeholder: string;
+  type: string;
+  required: boolean;
+}
+
+export const loader = (props: Props, req: Request) => {
+  const cookies = getCookies(req.headers);
+  const isOpen = !!cookies["DecoNewsletterModal"];
+
+  return { ...props, isOpen };
+};
+
+function InputNewsletter(
+  { name, placeholder, required, type }: InputNewletterProps,
+) {
+  return (
+    <input
+      name={name}
+      type={type}
+      class="input lg:h-12 h-9 px-5 join-item w-full mb-2.5 first:mt-5 border-2 border-neutral rounded-full placeholder:text-placeholder !outline-none lg:text-base text-xs"
+      placeholder={placeholder}
+      required={required}
+    />
+  );
+}
+
+function NewsletterModal(
+  {
+    isOpen,
+    form,
+    text,
+    modalSignExpiredDate,
+    modalCloseExpiredDate,
+  }: SectionProps<
+    ReturnType<typeof loader>
+  >,
+) {
+  const modalRef = useRef<HTMLDialogElement>(null);
+  const loading = useSignal(false);
+  const success = useSignal(false);
+
+  useEffect(() => {
+    if (isOpen !== true) {
+      modalRef.current?.showModal();
+    }
```

Comment:
Se `isOpen` for false, é pra aparecer o modal? Qual o sentido disso? Deveria ser 
```suggestion
    if (isOpen) {
      modalRef.current?.showModal();
    }
```